pkg/operator/ceph/pool/controller.go

```diff
@@ -456,27 +456,27 @@ func configureRBDStats(clusterContext *clusterd.Context, clusterInfo *cephclient
 		if cephBlockPool.GetDeletionTimestamp() == nil && cephBlockPool.Spec.EnableRBDStats {
 			// add to list of CephBlockPool with enableRBDStats set to true and not marked for deletion
 			enableStatsForCephBlockPools = append(enableStatsForCephBlockPools, cephBlockPool.ToNamedPoolSpec().Name)
+			logger.Infof("enableStatsForCephBlockPools_1=%q ", enableStatsForCephBlockPools)
 		}
 	}
 	enableStatsForCephBlockPools = remove(enableStatsForCephBlockPools, deletedPool)
+	logger.Infof("enableStatsForCephBlockPools_2=%q ", enableStatsForCephBlockPools)
 	monStore := config.GetMonStore(clusterContext, clusterInfo)
 	// Check for existing rbd stats pools
 	existingRBDStatsPools, e := monStore.Get("mgr", "mgr/prometheus/rbd_stats_pools")
 	if e != nil {
 		return errors.Wrapf(e, "failed to get rbd_stats_pools")
 	}
-
-	existingRBDStatsPoolsList := strings.Split(existingRBDStatsPools, ",")
-	enableStatsForPools := append(enableStatsForCephBlockPools, existingRBDStatsPoolsList...)
-	enableStatsForPools = removeDuplicates(enableStatsForPools)
-
-	logger.Debugf("RBD per-image IO statistics will be collected for pools: %v", enableStatsForPools)
-
-	if len(enableStatsForPools) == 0 {
+	logger.Infof("existingRBDStatsPools= %q", existingRBDStatsPools)
+	poolNames := strings.Trim(strings.Join(enableStatsForCephBlockPools, ","), ",")
+	logger.Infof("poolNames=%q ", poolNames)
+	if len(poolNames) == 0 {
+		logger.Infof("poolNames_1=%q ", poolNames)
```

Comment:
It seems like this will make it so pre-existing RBD stats pools will no longer be kept when Rook reconciles. That feature was added intentionally, so we should keep it. We just need to fix the bug that was also introduced at the same time that messed up removal of the current pool being reconciled.

Because this is a regression, we should also make sure to add a unit test to catch any future related regressions. The code to handle the combination/removal logic be added to a function which can then be thoroughly unit tested to ensure the proper behavior. I think talking about this function might help with the next code iteration as well.

The criteria for the function should be that:
1. stats pools from CephBlockPools (rookStatsPools) must be in the returned list
2. all existing Ceph stats pools (existingStatsPools) should be in the returned list
3. the exception to (2.) is that if a pool is being removed (removePool), it should not be in the returned list
4. the returned list should not contain duplicates
5. the returned list's ordering should be deterministic based on the inputs

Likely the function will have a signature something like `func generateStatsPoolList(existingStatsPools, rookStatsPools []string, removePool string) []string`

Reply:
done : tested on private image `oviner/rook:test7` [dockerhub]